app/models/legal_aid_application.rb

```diff
@@ -507,16 +507,27 @@ def hmrc_response_use_case_one
   end
 
   def uploading_bank_statements?
-    citizen_consent_given = provider_received_citizen_consent.nil? ? true : provider_received_citizen_consent?
-    provider.bank_statement_upload_permissions? && (!citizen_consent_given || attachments.bank_statement_evidence.exists?)
+    return false unless provider.bank_statement_upload_permissions?
+
+    !client_open_banking_consent? || attachments.bank_statement_evidence.exists?
```

Comment:
I made this change and introduced the private method because I found the existing method difficult to comprehend, and I felt like it didn't reveal the complexity (i.e the different conditional branches) clearly enough.

If people don't agree that it makes the method more readable, I can remove the change from this PR.

Reply:
it looks ok to me though my brain is a bit frazzled. But I believe unit tests cover this pretty well?!